core/src/main/java/org/elasticsearch/common/xcontent/NoContextParser.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent;
+
+import java.io.IOException;
+
+/**
+ * Reads an object from a parser without any context.
+ */
+@FunctionalInterface
+public interface NoContextParser<T> {
+    T parse(XContentParser p) throws IOException;
```

Comment:
this could even be a CheckedFunction<XContentParser, T, IOException> but maybe that makes things less readable?